### ISSUE TITLE
[Feature Request] Source Cross Referencing

### ISSUE THREAD

**Issue by SunBreaker1** [Jays2Kings/tachiyomiJ2K#862]
### Why/User Benefit/User Problem
With the large number of mangas saved from one source, it is difficult to track what mangas are saved from one source when browsing a second. This issue arose when MangaDex was hacked and taken down, resulting in me changing sources. With it now up I would like to have 2 main sources as MangaDex has alot of mangas that Mangalife doesn't.

### What/Requirements
Just like the manga migration feature allows the user to select which sources they want to migrate wo anoither source(s), this feature should work in a similar manner.

**Comment by CarlosEsco** [comment]
Can you explain what you are actually wanting?

**Comment by SunBreaker1** [comment]
> Can you explain what you are actually wanting?

Since the hack of MangaDex I migrated all my current Mangas over to MangaLife. Since then the MangaDex API has been brought back online and can be used to read. Because i like to use multiple sources for reading, as not all sources have the Mangas that i want and because i have a lot on my device saved to read. Simply put I was thinking of implementing a feature that allows the user to see what Mangas are already in the users library while browsing another. 

For example, if the user has 5 Mangas save from Mangalife, those Mangas would appear as "in library" while browsing another source, telling the user not to save it as it's already saved from another source. I would like to think that this feature could work by setting a main source and choosing one or more sources in which the mangas from the main source appear as already saved.

The attached pictures are what it could look like.
1st - current look
2nd - what I imagine

Sorry if I'm requesting a very specific feature that only I may want. Also, I apologize  if my explanation is lacking in details.

(I used paint, don't hate) 

![Cross save before](https://user-images.githubusercontent.com/85519676/121265669-26adda00-c887-11eb-932d-12d31b11d44f.jpg)
![Cross save after](https://user-images.githubusercontent.com/85519676/121265679-27df0700-c887-11eb-95a7-ad4bd45baeb3.jpg)

**Comment by 05k-44** [comment]
The only issue, that i can see, is that some extensions use the romanji name and some use the English release name, so it's hard to cross reference when it's looking for English but it's in romanji in your library

**Comment by SunBreaker1** [comment]
> The only issue, that i can see, is that some extensions use the romanji name and some use the English release name, so it's hard to cross reference when it's looking for English but it's in romanji in your library 

If this feature is implemented/worked on, there could be a work around where the user could manually either romanji or English. In reality this is a complex feature, no matter how simple this feature is, it could help the user experience as a quality of life improvement. And if this feature is liked, it be worked on more to work with both romanji and English.

I'm not a programmer nor do I know the inner  workings of this app and how it communicates with APIs and extensions, so I apologize if this feature is asking for too much.

**Comment by asdkant** [comment]
I also want this for the same reasons. One way to implement it could be:
- when adding to library, query mangaupdates or a similar manga database to try and find the manga
- save all the alternative names in a local DB
- when browsing manga not in the library, search the local DB and display matches under something saying "this manga seems to be in your collection already"

Description could also be added to the local DB for "already in collection" detection.

**Comment by Caio99BR** [comment]
Maybe it's related to https://github.com/Jays2Kings/tachiyomiJ2K/commit/0f50c30ad13e79959931ae486548a50c67da9248

**Comment by asdkant** [comment]
> this feature is implemented/worked on, there could be a work around where the user could manually either romanji or English. In reality this is a complex feature, no matter how simple this feature is, it could help the user experience as a quality of life improvement. And if this feature is liked, it be worked on more to work wi

Yes, the logic implemented there could definitely be used. But what is being proposed here is a bit more extensive, and it should run way more frequently. So maybe looking into optimizations would be a good idea.


Also, for NSFW sources it's common to get something along the lines of `[artist] title | alt title (series) {translator}` with other extra bits of information, so some smartness to detect that and match it to sources that just do regular titles would be appreciated (even if it's just for the detection done when adding to library)
